### PR TITLE
Possible fix for #7497: Allow weaker cipher suites

### PR DESCRIPTION
**Pull request by jonas-koeritz**
Maybe this fixes opencaching.de issues for some ancient android versions that do not support TLSv1.2 or any of the more modern cipher suites proposed by the server.
This explicitly allows all the cipher suites offered by opencaching.de and appends this weaker CipherSpec to the list of available specs to try.

### CODE REVIEW

**Comment by bekuno** [comment]
retest this please

**Comment by Lineflyer** [comment]
IMHO we should not put effort in solving this as long as we are not clear about the side effects on other Android versions/devices and other platforms.

In fact this problem seems limited to some 4.x devices trying to use opencaching.de. That sounds like a very limited amount of users.

**Comment by moving-bits** [comment]
I would rather not allow weaker cipher suites - don't like the idea of weakening security for that. If someone's still using such old (and insecure) devices they will have to live with some limitations, IMHO.

**Comment by jonas-koeritz** [comment]
This does not weaken the security for devices that support the higher security cipher suites significantly. It just allows weaker ciphers and doesn't force them.  I'm not a fan of supporting old Android devices forever. Maybe I should add this for older Android versions (<5) only. They don't have much to loose in terms of security anyways.

**Comment by SammysHP** [comment]
Most browsers remove support for TLS 1.0 and 1.1 this year, so I expect that many servers start to drop support for these versions as well.

**Comment by Lineflyer** [comment]
> Maybe I should add this for older Android versions (<5) only. They don't have much to loose in terms of security anyways.

From risk analysis perspective I would appreciate to limit the devices to only those old ones by a clear code passage (if <5 then), so the behaviour is unchanged on all other devices.

> Most browsers remove support for TLS 1.0 and 1.1 this year, so I expect that many servers start to drop support for these versions as well.

If this happens for gc.com (which a  major part of our users is using) this would be the time to just finally drop the support of Android 4.x as well.

**Comment by bekuno** [comment]
retest this please

**Comment by moving-bits** [comment]
Before merging, can the following be implemented?

> > Maybe I should add this for older Android versions (<5) only. They don't have much to loose in terms of security anyways.
> 
> From risk analysis perspective I would appreciate to limit the devices to only those old ones by a clear code passage (if <5 then), so the behaviour is unchanged on all other devices.

File: main/src/cgeo/geocaching/network/Network.java [hunk 1]
```diff
@@ -52,6 +53,8 @@
 import okhttp3.RequestBody;
 import okhttp3.Response;
 import okhttp3.TlsVersion;
+import okhttp3.internal.Util;
+
```
Comment (bekuno) [inline main/src/cgeo/geocaching/network/Network.java:57]: see https://ci.cgeo.org/job/cgeo%20pull%20request/2611/checkstyleResult/new/
remove extra separation in import group before 'org.apache.commons.compress.utils.IOUtils'

**Comment by Lineflyer** [comment]
Needs to be redone and retargeted to branch `cgeo-legacy` 
See #8602

**Pull request closed without merging**